terraform/variables.tf

```diff
@@ -43,12 +38,27 @@ variable "subnets" {
   description = "Subnets to use for deployment"
 }
 
-variable "tags" {
-  type        = map(string)
-  description = "Tags to use for AWS resources"
-}
-
 variable "vpc_id" {
   type        = string
   description = "VPC to use for deployment"
 }
+
+variable create_waf_acl {
+  type = bool
+  default = false
+}
+
+variable waf_acl_id {
+  type = string
+  default = ""
+  description = "The ACL to use with the Cloudfront distribution that is created. if not specified, an ACL which blocks all public access will be created"
+}
+
+variable kms_key_arn {
```

Comment:
I am wondering if we can create a default KMS key if this variable is not set (like you do for WAF ACL). It'd be one less manual step.

Reply:
In theory, you could do this, as long as you remembered to update Vault afterwards.

With this approach, the user can supply their own key they've created in a parent module.